compiler/src/codegen/compcore.re

```diff
@@ -965,14 +896,24 @@ let safe_drop = (wasm_mod, env, arg) =>
     call_decref_ignore_zeros(wasm_mod, env, arg),
   );
 
-let get_swap = (~ty as typ=I32Type, wasm_mod, env, idx) =>
+let get_swap =
+    (
+      ~ty as typ=I32Type,
+      ~skip_incref=true,
+      ~skip_decref=true,
```

Comment:
So, one of the changes I made was to really clamp down on the swap variables; since they should be treated as volatile registers and almost never interact with the refcounts. The swap variables only hold intermediate computation values, so we know their refcounts are valid during that computation.